site/content/posts/2021-06-21-windows-e2e.md

```diff
@@ -0,0 +1,134 @@
+
+## Getting A Windows Cluster
+
+If you would like to run Sonobuoy from a Windows node, you need to set up a Kubernetes cluster with Windows nodes. One of the easiest ways to get started is to use an Azure account and create an AKS cluster. It is easy but does require an account and costs based on resources used. Find more instructions [here](https://docs.microsoft.com/en-us/azure/aks/windows-container-cli).
+
+For the purposes of this blog post, you can start with a 2 node cluster:
+- 1 control plane, Linux node 
+- 1 Windows node for workloads
+
+We will discuss using a cluster with Linux workload clusters later in this post.
+
+## The First Milestone: Running A Single Test
+
+Before we run, we should walk. So we will start simply and run a single test that targets Windows behavior on this simplified cluster. Later, we will run more tests and include more complicated cluster configurations.
+
+Looking at the Kubernetes tests, let's choose a random test that targets Windows and runs quickly. One matching these goals is `should not be able to create pods with unknown usernames`.
+
+With our test chosen, lets move onto the next hurdle.
+
+## Problem One: E2E Image Compatibility
+
+If you try and run the existing e2e plugin on this cluster, the first problem you’ll run into is that the Sonobuoy aggregator will get placed onto the Windows node and run fine, but it will fail to launch the e2e plugin correctly since the only schedulable node is the Windows node (the Linux control plane node is tainted as NoSchedule by default) and the conformance image (which the e2e plugin uses) does not provide a Windows compatible image.
+
+To get around this, we are going to specify that the e2e plugin run on a Linux node. In our simple, 2-node cluster, this means that we need to add a nodeSelector to the podSpec as well as tolerate the control-plane NoSchedule taint (no need to actually run these commands, we are just discussing the logic right now).
+
+## Problem 2: Windows Skipped By Default
+
+Now if we were to run our chosen test with the tolerations/nodeSelectors we need, Sonobuoy will get placed on the Windows node and the e2e plugin will run from the Linux control-plane node. However, now you’ll run into another blocker: the tests are explicitly coded to skip any Windows test by default unless a flag is specified. Looking through the code, you can see that we just need to provide the `--node-os-distro=windows` flag when invoking the tests.
+
+If you include this value in the plugins environment variable `E2E_EXTRA_ARGS`, combined with the other points discussed above, you canb successfully run and pass a Windows specific test in your Kubernetes cluster.
+
+But running just one test on a simple test cluster isn't very useful. Let's talk about how we can run more tests on more complex clusters.
+
```

Comment:
At this point, after building up to milestone one, probably makes sense to provide some kind of code sample/cli commands to do what was discussed in milestone one (before moving on to milestone 2)